manifests/config.pp

```diff
@@ -7,6 +7,7 @@
   $remote_read_configs,
   $purge = true,
   $config_template = $::prometheus::params::config_template,
+  $retention = $::prometheus::params::storage_retention,
```

Comment:
You are referencing the variable from params.pp. Why not the variable from init.pp?
In your case the value is always fix and not changeable.

Reply:
I agree with @tuxmea here

Reply:
maybe I misunderstood your comment, but the parameter is changeable because it gets passed inside `init.pp`, in the instantiation of `class[::prometheus::config]` (and it's also a parameter to init itself).
and I wanted to stick to the current coding style, since `$config_template` is implemented in exactly the same way (AFAICT)

Reply:
@costela ok. I missed the parameter hand over in init.pp
Technically OK. Not my most favorite solution, but working as expected.

Reply:
oh, BTW, time for a bit of bikeshedding: `retention` or `storage_retention`? I couldn't decide ... :grin: